components/card/card.css

```diff
@@ -0,0 +1,253 @@
+.card {
+    font-size: 1.6rem;
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+    align-items: stretch;
+    max-width: 25rem;
+    min-height: 40rem;
+    max-height: 40rem;
+    border-radius: 10px;
+    box-shadow: 1px 1px 5px 1px var(--COLOR-SHADOW-2);
+    background-color: var(--WHITE);
+    transition: transform 0.2s ease-in, box-shadow 0.2s ease-in;
+    overflow: hidden;
+}
+
+.card:hover {
+    box-shadow: 1px 1px 5px 3px var(--COLOR-SHADOW-2);
+    box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
+    transform: scale(1);
+}
+
+/* card badge */
+
+.card-badge {
+    position: absolute;
+    font-size: 1.2rem;
+    text-transform: capitalize;
+    top: 10px;
+    padding: 5px 1rem;
+    background-color: var(--COLOR-WARNING-DARK);
+}
+
+/* card dismiss button */
+
+.card-dismiss-btn {
+    color: var(--GREY);
+    position: absolute;
+    top: 1rem;
+    right: 1rem;
+    padding: 0;
+}
+
+/* card text overlay */
+
+.card-text-overlay {
+    width: 100%;
+    height: 100%;
+    position: absolute;
+    background-color: var(--COLOR-SHADOW-3);
+    color: var(--WHITE);
+    display: flex;
+    align-items: center;
+    font-size: 2.2rem;
+    justify-content: center;
+    transition: visibility 0.2s ease-in, opacity 0.5s ease-in;
+    border-radius: 8px;
+    opacity: 0;
+    z-index: 2;
+}
+
+.card-text-overlay:hover {
+    opacity: 1;
+}
+
+/* card wishlist button */
+
+.card-favorite {
+    position: absolute;
+    top: 1rem;
+    right: 1rem;
+    padding: 0;
+    color: var(--GREY);
+}
+
+/* horizontal card */
+
+.card-horizontal {
+    flex-direction: row;
+    max-width: 45rem;
+    width: 100%;
+    min-height: 20rem;
+    max-height: 20rem;
+}
+
+.card-horizontal .card-img-wrapper {
+    max-width: 20rem;
+}
+
+.card-horizontal .card-price {
+    flex-direction: column;
+    gap: 0px;
+}
+
+.card-horizontal .card-discount-price {
+    font-size: 2.4rem;
+}
+
+.card-horizontal .card-cta {
+    flex-grow: 1;
+    justify-content: space-between;
+    flex-shrink: 0;
+}
+
+/* card list item */
+
+.card-list-item {
+    flex-direction: row;
+    min-width: 100%;
```

Comment:
any reason for min-width here?

Reply:
yup actually flex is making it shrink to smaller space I want to make it take the whole available space so used min-width here.

Reply:
in that case you can use flex-shrink: 0;